### PR TITLE
[ASTGen] Remove LegacyParser ping-pong mechanism

### PR DESCRIPTION
**Pull request by rintaro**
Now that `ASTGen` should be able to generate most Swift code. Let's remove "legacy parser" call-in, and remove the unhealthy cyclic dependency between lib/Parse and ASTGen.

### CODE REVIEW

File: lib/ASTGen/Sources/ASTGen/Decls.swift [hunk 1]
```diff
@@ -47,15 +47,16 @@ extension ASTGenVisitor {
     case .initializerDecl(let node):
       return self.generate(initializerDecl: node).asDecl
     case .macroDecl:
-      break
+      fatalError("unimplemented")
```
Comment (rintaro) [inline lib/ASTGen/Sources/ASTGen/Decls.swift:50]: Will rebase on #77412 after merging it.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by hamishknight — APPROVED** [review]
🎉

**Review by DougGregor — APPROVED** [review]
Fantastic progress!

**Pull request merged**